### ISSUE TITLE
Please make the timeout on a server a configurable option

### ISSUE THREAD

**Issue by derkgort** [UpCloudLtd/terraform-provider-upcloud#53]
Creating a or a bunch of servers in Singapore takes more time than 5 minutes.. Which is probably another issue. But it will finish, except not withing those 5 minutes and so it will not execute the provision parts.

https://github.com/UpCloudLtd/terraform-provider-upcloud/blob/master/upcloud/resource_upcloud_server.go#L209

**Comment by nikohaap** [comment]
It would be very nice to have this feature available. 
I have same kinds of problems with the upcloud_server_resource since it gives timeout error after 5 minutes. 

This error occurs if I try to create a bunch of servers in Amsterdam (e.g. 6 or more).

**Comment by nikohaap** [comment]
Yep, I have a same goal. It would be just nice to get the infrastructure up and running.

**Comment by derkgort** [comment]
I've just created a pull request for this. The timeout is changed from 5 to 25 minutes.
See #59

**Comment by hielee** [comment]
We've been having a lot of issues with instance creations timing out.

Can an Upcloud team member please review this?

cc @Darep @jgabor @Raiou @scop

**Comment by sebastiansterk** [comment]
Having the same issue but i already run into this timeout issue after 40 seconds.

**Comment by PopoSensei** [comment]
Hello and greetings from UpCloud.

I have reviewed this with the current master of the Provider, in which we have implemented http client which retries multiple times to create different resources with `retryablehttp "github.com/hashicorp/go-retryablehttp"` lib. This can be viewed in `terraform-provider-upcloud/upcloud/provider.go`

Tested this with starting 10 resources in Singapore, which is still a bit slow to start servers, especially from Finland, where I am located. It went mostly smoothly except one of the servers had Firewall resources tied to it, which didn't go up, since the server was in maintenance mode. No timeouts though, and the second apply did put up the Firewalls.

Is this still an issue for you, with the latest master?

**Comment by PopoSensei** [comment]
I will now close this, since I can't reproduce with the latest version. Feel free to open this issue again.

**Comment by boreBoar** [comment]
I run into this issue while creating more than 15 servers on UpCloud many a time. I'm attaching logs from my previous run below.
[run-ZNiXBTyNJ4DiViRL-apply-log.txt](https://github.com/UpCloudLtd/terraform-provider-upcloud/files/5397311/run-ZNiXBTyNJ4DiViRL-apply-log.txt)

I'm using
Terraform version: 0.12.29 on Terraform Cloud
UpCloud Terraform Provider: 0.1.0

In the logs I've launched 100 servers in region `us-sjo1` and the wait time has been past 1 hour till I had to cancel the run. 
From 100 servers, 32 were running at the end of the hour.
The others were indeed restarting after 25 minutes. What is the way to make sure the run completes successfully?
